src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -7166,9 +7166,7 @@ int SketchObject::addExternal(App::DocumentObject* Obj,
     std::vector<long> Types = ExternalTypes.getValues();
     std::vector<DocumentObject*> Objects = ExternalGeometry.getValues();
     std::vector<std::string> SubElements = ExternalGeometry.getSubValues();
-    if (Types.size() != Objects.size()) {
-        Types.resize(Objects.size(), 0);
-    }
+    Types.resize(Objects.size(), 0);

```

Comment:
```suggestion
    Types.resize(Objects.size(), static_cast<long>(ExtType::Projection));
```

If we're keeping to `std::vector<long>`, this should be made explicit. We could decide to use a different value for `ExtType::Projection` in the future, for example.